src/bridges/angle/AngleBridge.sol

```diff
@@ -0,0 +1,126 @@
+import {IStableMaster} from "../../interfaces/angle/IStableMaster.sol";
+
+/**
+ * @title Angle Protocol bridge contract.
+ * @author Angle Protocol Team
+ * @notice Allows a user to deposit/withdraw his tokens to/from Angle Protocol
+ * @dev Enter Angle Protocol as an SLP by depositing authorized tokens
+ */
+contract AngleBridge is BridgeBase {
+    using SafeERC20 for IERC20;
+
+    IStableMaster public constant STABLE_MASTER = IStableMaster(0x5adDc89785D75C86aB939E9e15bfBBb7Fc086A87);
+
+    // collateralAddress => poolManagerAddress
+    mapping(address => address) internal poolManagers;
+
+    // The amount of dust to leave in the contract
+    // Optimization based on EIP-1087
+    uint256 internal constant DUST = 1;
+
+    /**
+     * @notice Set address of rollup processor
+     * @param _rollupProcessor Address of rollup processor
+     */
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        poolManagers[0x6B175474E89094C44Da98b954EedeAC495271d0F] = 0xc9daabC677F3d1301006e723bD21C60be57a5915;
+        poolManagers[0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48] = 0xe9f183FC656656f1F17af1F2b0dF79b8fF9ad8eD;
+        poolManagers[0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2] = 0x3f66867b4b6eCeBA0dBb6776be15619F73BC30A2;
+        poolManagers[0x853d955aCEf822Db058eb8505911ED77F175b99e] = 0x6b4eE7352406707003bC6f6b96595FD35925af48;
+
+        _preApprove(0xc9daabC677F3d1301006e723bD21C60be57a5915);
+        _preApprove(0xe9f183FC656656f1F17af1F2b0dF79b8fF9ad8eD);
+        _preApprove(0x3f66867b4b6eCeBA0dBb6776be15619F73BC30A2);
+        _preApprove(0x6b4eE7352406707003bC6f6b96595FD35925af48);
+    }
+
+    /**
+     * @notice A function which returns an _totalInputValue amount of _inputAssetA
+     * @param _inputAssetA - ERC20/ETH (deposit), or sanToken (withdraw)
+     * @param _outputAssetA - sanToken (deposit), or ERC20/ETH (withdraw)
+     * @param _totalInputValue - amount of ERC20/ETH to deposit, or the amount of sanToken to withdraw
+     * @param _auxData - 0 (deposit), 1 (withdraw)
+     * @return outputValueA - the amount of output asset to return
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidInputA();
+        if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidOutputA();
+        if (_totalInputValue < 10) revert ErrorLib.InvalidInputAmount();
+
+        uint256 totalInputValue = _totalInputValue - DUST;
```

Comment:
Subtracting the DUST here would mean that the dust would keep on accumulating in the bridge. I think in this case it would be better to not subtract the value here and instead I would send the value in a separate transaction to the bridge after deployment. In the [other tests](https://github.com/AztecProtocol/aztec-connect-bridges/blob/80a8b3dbf1b2c5e302fc2613ec7588108269da34/src/test/bridges/liquity/StakingBridgeUnit.t.sol#L26) we simulate it by `dealing` that value to the bridge in the setup function.